llvm/lib/Target/AArch64/AArch64Subtarget.cpp

```diff
@@ -491,13 +491,15 @@ bool AArch64Subtarget::isStreamingCompatible() const {
 }
 
 bool AArch64Subtarget::isNeonAvailable() const {
-  return hasNEON() && !isStreaming() && !isStreamingCompatible();
+  if (hasSMEFA64())
+    return true;
+  return (hasNEON() && !isStreaming() && !isStreamingCompatible());
```

Comment:
You still need to check or `hasNEON()`. Is it simpler to write as:

```
return hasNEON() && 
       (hasSMEFA64() || (!isStreaming() && !isStreamingCompatible()));
```

Reply:
Done